11-Backbonejs/07-JavaScript-Objects-to-Backbone-Models.md

```diff
@@ -0,0 +1,230 @@
+
+As an example if we wanted to give a static function `sample()` to Scrabble we could do it this way:
+
+```javascript
+const Scrabble = Backbone.Model.extend(
+{  
+  sample: function() {
+    console.log("This works!");
+  }
+});
+```
+
+Notice that the function has the `name: function() { ...` syntax instead of `Scrabble.prototype.name = function() {...`.  We can convert the scoreLetter function this way.
+
+```javascript
+const Scrabble = Backbone.Model.extend(
+{
+  scoreLetter: function(letter) {
+    letter = letter.toUpperCase();
+    var letters = ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T', 'D', 'G', 'B', 'C', 'M', 'P', 'F', 'H', 'V', 'W', 'Y', 'K', 'J', 'X', 'Q', 'Z'];
+    var scores = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 5, 8, 8, 10, 10];
+    for (var i = 0; i < letters.length; i++){
+      if (letters[i] == letter){
+        return scores[i];
+      }
+    }
+    return 0;
+  }
+});
+```
+
+### Convert Scrabble.js to a Backbone Model.
+
+Now you & your partner work to convert the rest of the `scrabble.js` file to a Backbone Model.
+
+Then run the tests with `npm test`.
+
+### Check-In
+
+Now check your results with [this Gist](https://gist.github.com/CheezItMan/b8e420690c99b532a4399927480eb153).
+
+
+## Converting Player to Backbone
+
+Player is a bit more complicated, the player has some instance data, plays and name and a bunch of instance methods.  We'll take a two step approach, one step to get it working and then to improve our code and make it a bit more "Backbony." (tm).  
+
+### Getting the darn thing working!
+
+First we can convert the constructor to an `initialize()` function.
+
+```javascript
+const Player = Backbone.Model.extend({
+  defaults: {
+  },
+  initialize: function(options) {
+    this.name = options.name;
+    this.plays = [];
+    this.scrabble = new Scrabble();
+  },
+});
+```
+
+Notice that we changed this:
+
+```javascript
+const Player = function(name) {
+  this.name = name;
+  this.plays = [];
+  this.scrabble = new Scrabble();
+};
+```
+
+to this:
+
+```javascript
+initialize: function(options) {
+  this.name = options.name;
+  this.plays = [];
+  this.scrabble = new Scrabble();
+}
+```
+Because the initialize function plays much the same role as a constructor in Backbone we do many of the same actions in the initializemethod.  
+
+We also left the defaults object in, just so we can add to it later at the end.
+
+Before working on your own, you will also need to modify the `beforeEach()` function in the spec file to look like this:
+
+```javascript
+beforeEach(function() {
+  player = new Player({
+    name: "bob"
+});
+```
+
+### Now add the rest of the functions
+
+Next you and your partner finish migrating the rest of the functions into the Backbone Model.
+
+Then run your tests with `npm test`
+
+### Check-in
+
+When you finish you can check your code [here](https://gist.github.com/CheezItMan/3034de6f979b27e36144bf14b1088018):  
```

Comment:
This is pointing to a code sample with Player written as a plain JS object, was that intentional?